addons/web/static/tests/views/fields/many2many_tags_avatar_field_tests.js

```diff
@@ -604,4 +608,40 @@ QUnit.module("Fields", (hooks) => {
             );
         }
     );
+
+    QUnit.test(
+        "Many2ManyTagsAvatarField: make sure that the arch context is passed to the form view call",
+        async function (assert) {
+            serverData.views = {
+                "partner,false,form": `<form><field name="display_name"/></form>`,
+            };
+
+            const nameSearchProm = makeDeferred();
+            await makeView({
+                type: "list",
+                resModel: "turtle",
+                serverData,
+                arch: `<list editable="top"><field name="partner_ids" widget="many2many_tags_avatar" context="{ 'append_coucou': True }"/></list>`,
+                async mockRPC(route, args) {
+                    if (args.method === "name_search") {
+                        nameSearchProm.resolve();
+                    }
+                    if (args.method === "onchange" && args.model === "partner") {
+                        if (args.kwargs.context.append_coucou) {
+                            assert.step("onchange with context given");
+                        }
+                    }
+                },
+            });
+
+            await click(target.querySelector("div[name=partner_ids]"));
+            await editInput(target, `div[name="partner_ids"] input`, "A new partner");
+            await nameSearchProm;
```

Comment:
Why ? Maybe you just need to patch settimeout ?
            patchWithCleanup(browser, {
                setTimeout: (fn) => fn(),
            });

Reply:
yeah you are right. I used the ``many2many_tags_field`` tests as inspiration for that one and they were using that so I used it as well. 
Indeed patching ``setTimeout`` also works, I changed it to that

Reply:
Nice :) It s a better practice.